### PR TITLE
Fix a bug in FE_Face.

### PR DESCRIPTION
**Pull request by bangerth**
Found while looking at #13150. The fix is obvious from the context a few lines above. I'm curious if that will break anything.

/rebuild

### CODE REVIEW

File: source/fe/fe_face.cc [hunk 1]
```diff
@@ -956,7 +956,7 @@ FE_FaceP<dim, spacedim>::get_subface_interpolation_matrix(
           for (unsigned int k = 0; k < face_quadrature.size(); ++k)
             {
               const Point<dim - 1> p =
-                numbers::invalid_unsigned_int != 0u ?
+                subface != numbers::invalid_unsigned_int ?
```
Comment (drwells) [inline source/fe/fe_face.cc:959]: How the heck did we not get a tautological comparison warning here?

Comment (drwells) [reply source/fe/fe_face.cc:959]: Also - I think this should be `==` here, like on line 938.

Comment (bangerth) [reply source/fe/fe_face.cc:959]: So much for it being obvious :-( I fixed it, thanks for noticing!

**Comment by bangerth** [comment]
I updated the output. I convinced myself that the previous output must have been wrong and I believe the current one is correct.

**Review by masterleinad — APPROVED** [review]
Thanks for fixing this!

**Comment by bangerth** [comment]
Thanks for finding it for us!

**Pull request merged**